streaming/base/partition/relaxed.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Apportion shards/samples to nodes/ranks/workers for elastically deterministic sample order."""
+
+import logging
+from typing import Optional
+
+import numpy as np
+from numpy.typing import NDArray
+
+from streaming.base.partition.orig import get_partitions_orig
+
+logger = logging.getLogger(__name__)
+
+
+def get_partitions_relaxed(num_samples: int,
```

Comment:
Can we add some `logger.debug()` messages in this method and other places that display the shapes or any additional helpful information when a user enables the logger to debug mode?

Reply:
Based on the offline discussion, let's track adding a logger.debug messages in this repo as part of another PR.